app/models/campaign_pages_widget.rb

```diff
@@ -1,12 +1,11 @@
 class CampaignPagesWidget < ActiveRecord::Base
 
-  belongs_to :campaign_page
+  belongs_to :campaign_page, inverse_of: :campaign_pages_widgets
   belongs_to :widget_type
   has_one :actionkit_page
 
   validates_presence_of :content, :page_display_order, :campaign_page_id, :widget_type_id
   # validates that there are not two widgets with exactly same content on the same campaign page
```

Comment:
I'm not totally sure that we want to validate on this combination. I'm not sure we don't, either; I just think it's worth thinking a bit about and talking it over with Paul and Eoin to get a feel for what we'd want to do here.